src/test/java/org/opensearch/flowframework/FlowFrameworkRestTestCase.java

```diff
@@ -676,6 +676,38 @@ protected SearchResponse searchWorkflowState(RestClient client, String query) th
         }
     }
 
+    /**
+     * Helper method to invoke the Search Agent Rest Action
+     * @param client the rest client
+     * @param query the search query
+     * @return
+     * @throws Exception
+     */
+    protected SearchResponse searchAgent(RestClient client, String query) throws Exception {
+        Response restSearchResponse = TestHelpers.makeRequest(
+            client,
+            "GET",
+            "/_plugins/_ml/agents/_search",
+            Collections.emptyMap(),
+            query,
+            null
+        );
+        assertEquals(RestStatus.OK, TestHelpers.restStatus(restSearchResponse));
+
+        // Parse entity content into SearchResponse
+        MediaType mediaType = MediaType.fromMediaType(restSearchResponse.getEntity().getContentType());
```

Comment:
The test can be a different method and `searchAgent` can include just the API call.

Reply:
Not sure what you mean. I'll change searchAgent to getAgent. The code will be similar to `getWorkflow`.

Reply:
I see searchAgent() being used in `src/test/java/org/opensearch/flowframework/rest/FlowFrameworkRestApiIT.java`, so this can be ignored.